app/Actions/Photo/Archive.php

```diff
@@ -90,8 +90,8 @@ protected function file(Photo $photo, DownloadVariantType $downloadVariant): Str
 
 		$responseGenerator = function () use ($archiveFileInfo) {
 			$outputStream = fopen('php://output', 'wb');
-			stream_copy_to_stream($archiveFileInfo->getFile()->read(), $outputStream);
-			$archiveFileInfo->getFile()->close();
+			stream_copy_to_stream($archiveFileInfo->file->read(), $outputStream);
```

Comment:
I had a doubt, but because ArchiveFileInfo is now `final` and `readonly` I switched it to public.
https://github.com/LycheeOrg/Lychee/blob/9deb15b5a4f5fbeb5a47d9681c13542109afa9db/app/Actions/Photo/Extensions/ArchiveFileInfo.php#L50-L54

Reply:
Read-only properties have some limitations due to they do not ensure values are inmutable.

They only ensure that is not possible to assign a new value.

<https://php.watch/versions/8.1/readonly#limitations>

Using getters avoids that problem for strings and arrays (as the are copied), but not for objects (as they are passed bay reference).

That said, mos of the cases you can use  public read-only properties as replacement of getters with no problems.